File: OfflineFetcher/app/src/main/java/de/jbamberger/offlineclient/ui/reddit/RedditViewModel.java
package de.jbamberger.offlineclient.ui.reddit;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import javax.inject.Inject;

import de.jbamberger.offlineclient.db.entity.RedditPost;
import de.jbamberger.offlineclient.source.reddit.RedditRepository;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

public class RedditViewModel extends ViewModel {
    private LiveData<RedditPost> post;
    private RedditRepository repo;

    @Inject
    public RedditViewModel(RedditRepository repo) {
        this.repo = repo;
    }


    public void init(String subreddit) {
        if (this.post != null) {
            return;
        }
        post = repo.getRedditPost(subreddit);
    }
    public LiveData<RedditPost> getPost() {
        return post;
    }
}

File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/ui/reddit/RedditViewModel.kt
package de.jbamberger.offlineclient.ui.reddit

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel
import de.jbamberger.api.reddit.RedditPost
import de.jbamberger.api.reddit.RedditRepository
import javax.inject.Inject

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

class RedditViewModel @Inject
constructor(private val repo: RedditRepository) : ViewModel() {
    var post: LiveData<RedditPost>? = null
        private set


    fun init(subreddit: String) {
        if (this.post != null) {
            return
        }
        post = repo.getRedditPost(subreddit)
    }
}
